src/pages/articles/canvas/animation/index.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Animation
+---
+## Animation in Canvas
+
+To animate things in `canvas`, use `window.requestAnimationFrame` to set up a draw loop.
+
+```
+function draw() {
+  /* code goes here */
+  window.requestAnimationFrame(draw);
+}
+window.requestAnimationFrame(draw);
+```
+
+The below code will cause the `draw` function to be run every frame.
+
+`canvas` has no special functions that allow for animating. You just have to be used to writing in animation loops. The usual design paradigm for animation loops is to update the state, then draw the state. For instance, to draw a square moving across the screen:
+
+```
+var x=0;
+var y=50;
+function draw() {
+  // reset canvas
+  ctx.clearRect(0, 0, canvas.width, canvas.height);
```

Comment:
In this case, I thought that the `canvas = document.getElementById("canvas")` and `ctx = canvas.getContext('2d')` calls are so common that I didn't want to repeat them over and over again in each article. Should I add them in anyway?